corehq/apps/hqcase/api.py

```diff
@@ -0,0 +1,210 @@
+import uuid
+
+from memoized import memoized
+
+import jsonobject
+from jsonobject.exceptions import BadValueError
+
+from casexml.apps.case.mock import CaseBlock, IndexAttrs
+
+from corehq.apps.hqcase.utils import submit_case_blocks
+from corehq.form_processor.interfaces.dbaccessors import CaseAccessors
+
+
+def serialize_case(case):
+    """Serializes a case for the V0.6 Case API"""
+    return {
+        "domain": case.domain,
+        "@case_id": case.case_id,
+        "@case_type": case.type,
+        "case_name": case.name,
+        "external_id": case.external_id,
+        "@owner_id": case.owner_id,
+        "date_opened": _isoformat(case.opened_on),
+        "last_modified": _isoformat(case.modified_on),
+        "server_last_modified": _isoformat(case.server_modified_on),
+        "closed": case.closed,
+        "date_closed": _isoformat(case.closed_on),
+        "properties": dict(case.dynamic_case_properties()),
+        "indices": {
+            index.identifier: {
+                "case_id": index.referenced_id,
+                "@case_type": index.referenced_type,
+                "@relationship": index.relationship,
+            }
+            for index in case.indices
+        }
+    }
+
+
+def _isoformat(value):
+    return value.isoformat() if value else None
```

Comment:
Is value always guaranteed to be a date? If yes, this is fine, otherwise
```suggestion
    return value.isoformat() if isinstance(value, datetime) else None
```
or possibly
```suggestion
    return value if value is None or isinstance(value, str) else value.isoformat()
```

Reply:
Yes, it should be either a date or `None` - it's stored as a nullable `DateTimeField`.  I don't plan to use this for any user-defined properties - those I'll leave as strings.

I will also need to serialize the version of cases stored in ES, but I'm guessing that'll be a different function entirely, given the differences between the two representations.